quinn-proto/src/crypto/rustls.rs

```diff
@@ -386,10 +395,14 @@ impl std::error::Error for NoInitialCipherSuite {}
 
 /// A QUIC-compatible TLS server configuration
 ///
-/// Can be constructed via [`ServerConfig::with_single_cert()`][single] or by using the
-/// [`TryFrom`] implementation with a custom [`rustls::ServerConfig`]. A pre-existing
-/// `ServerConfig` must have TLS 1.3 support enabled for this to work. 0-RTT support is
-/// available to clients if `max_early_data_size` is set to `u32::MAX`.
+/// Quinn implicitly constructs a `QuicServerConfig` with reasonable defaults within
+/// [`ServerConfig::with_single_cert()`][single]. Alternatively, `QuicServerConfig`'s [`TryFrom`]
+/// implementation or `with_initial` method can be used to wrap around a custom
+/// [`rustls::ServerConfig`], in which case care should be taken around certain points:
+///
+/// - If `max_early_data_size` is not set to `u32::MAX`, the server will not be able to accept
+///   incoming 0-RTT data. QUIC prohibits `max_early_data_size` values other than 0 or `u32::MAX`.
```

Comment:
```suggestion
///   incoming 0-RTT data. QUIC prohibits `max_early_data_size` values other than disabled or `u32::MAX`.
```
Strictly speaking, RFC 9001 [forbids](https://www.rfc-editor.org/rfc/rfc9001.html#section-4.6.1-2) *any* value other than `u32::MAX`; to disable it, the extension is not sent at all. It's an idiosyncrasy of rustls that setting the value to 0 means not sending it.

Reply:
Eh, we're documenting the rustls config here, this is fine.